src/windows/text_input.c

```diff
@@ -249,7 +249,11 @@ static void window_text_input_paint(){
 			}
 
 			if (w->frame_no > 15){
-				gfx_fill_rect(dpi, cur_x, y + 9, cur_x + width, y + 9, w->colours[1]);
+				uint8 eax;
+				eax = RCT2_ADDRESS(0x0141FC4A, uint8)[w->colours[1] * 8];
+				eax = eax << 16;
+				eax = eax | RCT2_ADDRESS(0x0141FC48, uint8)[w->colours[1] * 8];
+				gfx_fill_rect(dpi, cur_x, y + 9, cur_x + width, y + 9, (eax & 0xFF) + 5);
```

Comment:
I don't understand why you bother filling the top word of the eax with a value and then promptly not using it. Simplify this to use a uint8 variable called colour and then it will make more sense.